kt/godot-runtime/src/main/kotlin/godot/core/TransferContext.kt

```diff
@@ -21,7 +23,7 @@ object TransferContext {
 
     private var inputStream  = object : InputStream() {
         override fun read(): Int {
-            return buffer.get().toInt()
+            return buffer.get().toInt() and 0xFF
```

Comment:
hmmm, do we still need the `toInt`?

Reply:
Yep we still need it I think.
The problem was that `toInt` returns an int with same significant 8 first bit, so I think mask should be applied on them after int conversion.

Reply:
but we lose the MSB for the byte right?

Reply:
Maybe I'm blind on this problem but think this should not.
If I take -2 signed byte as an example:
Its unsigned equivalent is 254, which is :
1111 1110
When converting it to int, it will create -2 int which is : 
1111 1111 1111 1111 1111 1111 1111 1111 1111 1111 1111 1111 1111 1111 1111 1110
When applying 255 mask, we only keep 1111 1110, which result in 254.
As values goes from -128 to 127 this should not cause problem. 

But anyway, maybe we should use UByte class as you suggested on discord, as it would clearly express intention.